From 79c943bba37cb10a462038ec56cd2b8045344951 Mon Sep 17 00:00:00 2001
From: Donn Seeley <donn.seeley@windriver.com>
Date: Fri, 28 Feb 2014 09:11:02 -0500
Subject: [PATCH] Add the glusterfs package

Three of the translator makefiles in glusterfs add unnecessary
-L$(xlatordir) link options.  This option causes the linker to check
the host's $(xlatordir) directory, resulting in contamination that gets
flagged by Yocto's QA rules.

Upstream-Status: Pending

Signed-off-by: Donn Seeley <donn.seeley@windriver.com>
[Xulin: minor adjust to apply the patch]
Signed-off-by: Xuin Sun <xulin.sun@windriver.com>

---
 xlators/mgmt/glusterd/src/Makefile.am    | 2 +-
 xlators/nfs/server/src/Makefile.am       | 2 --
 xlators/system/posix-acl/src/Makefile.am | 2 --
 3 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/xlators/mgmt/glusterd/src/Makefile.am b/xlators/mgmt/glusterd/src/Makefile.am
index b0f5a9b..0caad43 100644
--- a/xlators/mgmt/glusterd/src/Makefile.am
+++ b/xlators/mgmt/glusterd/src/Makefile.am
@@ -55,7 +55,7 @@ AM_CPPFLAGS = $(GF_CPPFLAGS) -I$(top_srcdir)/libglusterfs/src \
 
 AM_CFLAGS = -Wall $(GF_CFLAGS) $(URCU_CFLAGS) $(URCU_CDS_CFLAGS)
 
-AM_LDFLAGS = -L$(xlatordir) $(URCU_LIBS) $(URCU_CDS_LIBS)
+AM_LDFLAGS = -L$(URCU_LIBS) $(URCU_CDS_LIBS)
 
 CLEANFILES =
 
diff --git a/xlators/nfs/server/src/Makefile.am b/xlators/nfs/server/src/Makefile.am
index bc6f8e9..56086a1 100644
--- a/xlators/nfs/server/src/Makefile.am
+++ b/xlators/nfs/server/src/Makefile.am
@@ -25,8 +25,6 @@ AM_CPPFLAGS = $(GF_CPPFLAGS) \
 
 AM_CFLAGS = -Wall $(GF_CFLAGS)
 
-AM_LDFLAGS = -L$(xlatordir)
-
 CLEANFILES =
 
 EXTRA_DIST = nfsserver.sym
diff --git a/xlators/system/posix-acl/src/Makefile.am b/xlators/system/posix-acl/src/Makefile.am
index f57070c..70080a6 100644
--- a/xlators/system/posix-acl/src/Makefile.am
+++ b/xlators/system/posix-acl/src/Makefile.am
@@ -12,8 +12,6 @@ AM_CPPFLAGS = $(GF_CPPFLAGS) -I$(top_srcdir)/libglusterfs/src \
 
 AM_CFLAGS = -Wall $(GF_CFLAGS)
 
-AM_LDFLAGS = -L$(xlatordir)
-
 CLEANFILES =
 
 access-control-compat:
